src/inmemory.js

```diff
@@ -0,0 +1,367 @@
+ *   top:               1000,          // Max number of entities to return
+ *
+ *   // Paging options:
+ *   nextPartitionKey:  '...',         // nextPartitionKey from previous result
+ *   nextRowKey:        '...'          // nextRowKey from previous result
+ * }
+ * ```
+ * @return {Promise} A promise for an object on the form:
+ * ```js
+ * {
+ *   entities: [
+ *     {
+ *       // Keys selected from entity and meta-data depending on meta-data level
+ *     },
+ *     ...
+ *   ],
+ *   nextPartitionKey: '...',  // Opaque token for paging
+ *   nextRowKey:       '...'   // Opaque token for paging
+ * }
+ * ```
+ */
+InMemoryWrapper.prototype.queryEntities = function(options) {
+  options = options || {};
+  assert(!options.select, "select is not supported for queryEntities");
+
+  if (!tables[this.table]) {
+    return Promise.reject(makeError(404, 'ResourceNotFound'));
+  }
+  
+  var filterFunctionFor = function(filter) {
+    return function(entity) {
+      return _.every(filter, function(condition) {
+        return condition.type.compare(entity, condition.op);
+      });
+    };
+  };
+
+  var entities = _.values(tables[this.table]);
+
+  if (options.filter) {
+    entities = _.filter(entities, filterFunctionFor(options.filter));
+  }
+
+  if (options.nextRowKey || options.nextPartitionKey) {
+    var rowKey = options.nextRowKey,
+        partitionKey = options.nextPartitionKey;
+    while (entities.length > 0 &&
+           (entities[0].PartitionKey != partitionKey ||
+            entities[0].RowKey != rowKey)) {
+      entities.shift();
+    }
+  }
+
+  var nextPartitionKey, nextRowKey;
+  if (options.top) {
+    if (entities.length > options.top) {
+      nextPartitionKey = entities[options.top].PartitionKey;
+      nextRowKey = entities[options.top].RowKey;
+    }
+    entities = entities.slice(0, options.top);
+  }
+
+  // Apply pagination
+  return Promise.resolve({
+    entities: entities,
+    nextPartitionKey: nextPartitionKey,
+    nextRowKey: nextRowKey,
+  });
+};
+
+/**
+ * Insert `entity`. The `entity` object must be on the format
+ * accepted by azure table storage. See Azure Table Storage documentation for
+ * details. Essentially, data-types will be inferred if `...@odata.type`
+ * properties aren't specified. Also note that `PartitionKey` and `RowKey`
+ * properties must be specified.
+ *
+ * @method insertEntity
+ * @param {object} entity - Entity object, see Azure Table Storage
+ * documentation for details on how to annotate types.
+ * @return {Promise}
+ * A promise for the `etag` of the inserted entity.
+ */
+InMemoryWrapper.prototype.insertEntity = function(entity) {
+  var key = makeKey(entity.PartitionKey, entity.RowKey);
+  if (!tables[this.table]) {
+    return Promise.reject(makeError(404, 'ResourceNotFound'));
+  }
+  if (key in tables[this.table]) {
+    return Promise.reject(makeError(409, 'EntityAlreadyExists'));
+  }
+  updateTimestamp(entity);
```

Comment:
Probably best to do an `entity = _.cloneDeep(entity)` before storing it... Probably the same in modify...